### ISSUE TITLE
1.3.2 remote forward failure

### ISSUE THREAD

**Issue by hrenki** [ThomDietrich/home-assistant-addons#23]
Hi,
after update to 1.3.2 cannot make connection to remote host

debug1: Remote connections from 127.0.0.1:8123 forwarded to local address 172.30.32.1:8123
debug1: Remote connections from LOCALHOST:8123 forwarded to local address 127.0.0.1:8123
...
debug1: remote forward success for: listen 127.0.0.1:8123, connect 172.30.32.1:8123
debug1: remote forward failure for: listen 8123, connect 127.0.0.1:8123

i see that on 1.2.1 is only debug1: Remote connections from LOCALHOST:8123 forwarded to local address 127.0.0.1:8123 and it works

**Comment by funesbubble** [comment]
The same issue occurred. It was working well yesterday, but after the upgrade, it failed.
Configuration:
hostname: 159.226.221.42
ssh_port: 22
username: reverse-tunnel
remote ip: 127.0.0.1
remote port: 8123
remote_forwarding: 159.226.221.42:8123:localhost:8123
other_ssh_options: -v -N -M 0 -o LogLevel=error -o UserKnownHostsFile=/dev/null -o     StrictHostKeyChecking=no -o ExitOnForwardFailure=yes -o     ServerAliveInterval=10 -o ServerAliveCountMax=3
--
LOG on HA side:
debug1: Remote: Forwarding listen address "127.0.0.1" overridden by server GatewayPorts
debug1: remote forward success for: listen 127.0.0.1:8123, connect 172.30.32.1:8123
debug1: Remote: Forwarding listen address "159.226.221.42" overridden by server GatewayPorts
debug1: remote forward failure for: listen 159.226.221.42:8123, connect localhost:8123
Error: remote port forwarding failed for listen port 8123
--
LOG on remote server:
Oct 18 10:40:24 ubuntu-01 systemd[1]: Started Session 71 of User reverse-tunnel.
Oct 18 10:40:25 ubuntu-01 systemd[1]: session-71.scope: Deactivated successfully.

**Comment by funesbubble** [comment]
The same issue occurred. It was working well yesterday, but after the upgrade, it failed.
---
Configuration:
hostname: 159.226.221.42
ssh_port: 22
username: reverse-tunnel
remote ip: 127.0.0.1
remote port: 8123
remote_forwarding: 159.226.221.42:8123:localhost:8123
other_ssh_options: -v -N -M 0 -o LogLevel=error -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o ExitOnForwardFailure=yes -o ServerAliveInterval=10 -o ServerAliveCountMax=3
---
LOG on HA side:
debug1: Remote: Forwarding listen address "127.0.0.1" overridden by server GatewayPorts
debug1: remote forward success for: listen 127.0.0.1:8123, connect 172.30.32.1:8123
debug1: Remote: Forwarding listen address "159.226.221.42" overridden by server GatewayPorts
debug1: remote forward failure for: listen 159.226.221.42:8123, connect localhost:8123
Error: remote port forwarding failed for listen port 8123
---
LOG on remote server:
Oct 18 10:40:24 ubuntu-01 systemd[1]: Started Session 71 of User reverse-tunnel.
Oct 18 10:40:25 ubuntu-01 systemd[1]: session-71.scope: Deactivated successfully.

**Comment by funesbubble** [comment]
The same issue occurred. It was working well yesterday, but after the upgrade, it failed.
Configuration:

> hostname: 159.226.221.42
> ssh_port: 22
> username: reverse-tunnel
> remote ip: 127.0.0.1
> remote port: 8123
> remote_forwarding: 159.226.221.42:8123:localhost:8123
> other_ssh_options: -v -N -M 0 -o LogLevel=error -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o ExitOnForwardFailure=yes -o ServerAliveInterval=10 -o ServerAliveCountMax=3

LOG on HA side:

> debug1: Remote: Forwarding listen address "127.0.0.1" overridden by server GatewayPorts
> debug1: remote forward success for: listen 127.0.0.1:8123, connect 172.30.32.1:8123
> debug1: Remote: Forwarding listen address "159.226.221.42" overridden by server GatewayPorts
> debug1: remote forward failure for: listen 159.226.221.42:8123, connect localhost:8123
> Error: remote port forwarding failed for listen port 8123
> 

LOG on remote server:

> Oct 18 10:40:24 ubuntu-01 systemd[1]: Started Session 71 of User reverse-tunnel.
> Oct 18 10:40:25 ubuntu-01 systemd[1]: session-71.scope: Deactivated successfully.

**Comment by ThomDietrich** [comment]
Hey there. This is likely due to your carried over configuration.

Your addon configuration should look similar to this now:

```yaml
hostname: ssh.yourdomain.com
ssh_port: 2222
username: homeassistant
remote_forwarding: []
remote_ip_address: 127.0.0.1
remote_port: 8000
other_ssh_options: ""
force_keygen: false
```

Which is equivalent to the old config style of:

```yaml
remote_forwarding:
  - 127.0.0.1:8000:172.30.32.1:8123
```

**Comment by ThomDietrich** [comment]
Please check if your issue is resolved with 1.3.3. Thanks!

**Comment by hrenki** [comment]
didnt work straight after upgrade, but when edited yaml there was a value in remote_forwarding: and after deleting value and setting it to [] all works
